### PR TITLE
Refactor/playlist

### PR DESCRIPTION
**Pull request by Somil680**

### CODE REVIEW

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 1]
```diff
@@ -1,59 +1,64 @@
-import React , {useState} from 'react';
+import React, { useState } from 'react';
+
 import { usePlaylist } from '../../context';
-import {v4 as uuid} from "uuid"
+import { v4 as uuid } from "uuid"
+
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:5]: move to 2 line

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 2]
```diff
@@ -1,59 +1,64 @@
...
-}
-// YE FUNCTION EK NAYI VIDEO KO EXISTING PLAYLIST ME DALTA HAI ! 
-const addMoreInPlaylist = (index, item) => {
-    const isInPlaylist = playlistState.Playlistitem.find((ele)=> ele.ID === item.ID)
+    }
+    // YE FUNCTION ADD BUTTON PE CLICK HONE SE TIGGERED HOTA AUR SAARA DATA PLAYLIST ME SAVE HOJATA HAI !
+    const addInPlaylist = () => {
+        playlistDispatch({
+            type: "ADD-ITEM",
+            payload: { ID: uuid(), name, Videos: [Newvideo] }
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:22]: ADD_ITEM

Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:22]: id

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 3]
```diff
@@ -1,59 +1,64 @@
...
+        })
+    }
+    // YE FUNCTION EK NAYI VIDEO KO EXISTING PLAYLIST ME DALTA HAI !
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:25]: in english

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 4]
```diff
@@ -1,59 +1,64 @@
...
+    const addMoreInPlaylist = (index, item) => {
+        const isInPlaylist = playlistState.Playlistitem.find((ele) => ele.ID === item.ID)
         if (isInPlaylist) {
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:28]: don't use inline function
ele => element

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 5]
```diff
@@ -1,59 +1,64 @@
...
             playlistDispatch({
                 type: "ADD-MORE-ITEM",
-                payload:{Videos: [Newvideo]},
-                index : index
+                payload: { Videos: [Newvideo] },
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:31]: ___

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 6]
```diff
@@ -1,54 +1,57 @@
 import React from 'react';
-import "./Videoshow.scss"
+
 import { findInArray } from '../../Utilis/find';
 import { useHistory } from '../../context';
-import { HiDotsVertical } from "react-icons/hi";
 import { Link } from 'react-router-dom';
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx]: move to 2 line

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 7]
```diff
@@ -1,54 +1,57 @@
...
-function historyHandler() {
-    if (!isInHistory) {  
-        historyDispatch({
-            type: "ADD-TO-HISTORY",
-            payload : videos
-        }) 
+import "./RecommendVideo.scss"
+
+function RecommendVideo({ videos }) {
+    const { _id, title, iframeId, publishedAt, creator } = videos
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:10]: dont destructure here

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 8]
```diff
@@ -1,54 +1,57 @@
...
+    const { historyState, historyDispatch } = useHistory()
+    const { historyItem } = historyState
+
+    const isInHistory = findInArray(_id, historyItem)
+    function historyHandler() {
+        if (!isInHistory) {
+            historyDispatch({
+                type: "ADD-TO-HISTORY",
+                payload: videos
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:19]: ___

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 9]
```diff
@@ -1,54 +1,57 @@
...
-}
 
-const styles = {
-        recommend_Video: {
-        sub_content: "bfy__rec_video_sub_content",
-        tumbnail: "bfy__rec_video_tumbnail",
-        details : "bfy__rec_video_details",
+    const styles = {
+        recommendVideo: {
+            discription: "bfy__videoshow__recommendVideo-discription",
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:26]: recommended-video

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 10]
```diff
@@ -1,54 +1,57 @@
...
 
 
 
-const getRecommend_cardView = () => (
-    <Link key={_id} to={`/watch=/${_id}`} className="Link" onClick={()=>historyHandler()}>
-        <div key={_id} className={styles.recommend_Video.sub_content}>
-            <img src={`https://img.youtube.com/vi/${iframeId}/maxresdefault.jpg `} alt="" className={styles.recommend_Video.tumbnail}/>
+    const getRecommend_cardView = () => (
+        <Link key={_id} to={`/watch=/${_id}`} className="Link" onClick={() => historyHandler()}>
+            <div key={_id} className={styles.recommendVideo.discription}>
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:38]: dont use inline function

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 11]
```diff
@@ -1,54 +1,57 @@
...
+                <img src={`https://img.youtube.com/vi/${iframeId}/maxresdefault.jpg `} alt="" className={styles.recommendVideo.thumbnail} />
                 <div>
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:40]: move to statics folder

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 12]
```diff
@@ -1,54 +1,57 @@
...
+                        <p className={styles.recommendVideo.text}>{creator}</p>
+                        <p className={styles.recommendVideo.text}>{publishedAt}</p>
+                    </span>
+                </div>
+
+            </div>
+        </Link>
+    )
+
+    return (<>
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:52]: don't use fragment, use div

File: src/Pages/Videoshow/RecommendVideo.jsx [hunk 13]
```diff
@@ -1,54 +1,57 @@
...
+        {getRecommend_cardView()}
+
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.jsx:54]: remove _

File: src/Pages/Videoshow/RecommendVideo.scss [hunk 14]
```diff
@@ -0,0 +1,67 @@
+@media screen and (min-width: 401px) {
+    .bfy__videoshow {
+        &__recommendVideo {
+            &-discription {
+                margin: 0px 20px;
+                display: flex;
+                padding: 0px 0px 12px 0px;
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.scss:7]: padding-bottom

File: src/Pages/Videoshow/RecommendVideo.scss [hunk 15]
```diff
@@ -0,0 +1,67 @@
+@media screen and (min-width: 401px) {
+    .bfy__videoshow {
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.scss:2]: move to rem

File: src/Pages/Videoshow/RecommendVideo.scss [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+            }
+        }
+    }
+}
+@media screen and (max-width: 400px) {
+    .bfy__videoshow {
+        &__recommendVideo {
+            &-discription {
+                padding: 0px 0px 31px 0px;
+                display: flex;
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/RecommendVideo.scss:39]: padding-bottom

File: src/Pages/Videoshow/Videoshow.jsx [hunk 17]
```diff
@@ -1,58 +1,54 @@
...
+import "./Videoshow.scss"
 function Videoshow() {
 
     const { videoId } = useParams()
     const { videoState } = useVideos()
     const videoShow = videoState.videos
     const videos = [videoShow.find((element) => element._id === videoId)];
-    const videotag = videos.map((videostged) => videostged.tag)
+    const videotag = videos.map((tags) => tags.tag)
     const tag = videotag.join()
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/Videoshow.jsx:19]: move to inline

File: src/Pages/Videoshow/Videoshow.jsx [hunk 18]
```diff
@@ -1,58 +1,54 @@
...
     const tagVideos = videoShow.filter((element) => element.tag === tag);
     const Finalrecommendvideo = tagVideos.filter((ele) => ele._id !== videoId)
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/Videoshow.jsx:22]: ele ky hota hai?  make it insense

Comment (agrawalshanu708) [inline src/Pages/Videoshow/Videoshow.jsx:22]: remove inline function

File: src/Pages/Videoshow/Videoshow.jsx [hunk 19]
```diff
@@ -1,58 +1,54 @@
...
 
     const styles = {
-        header: "bfy__videoplayer__header",
-        video_Player: {
-            content : "bfy__video_Player__content"
+        header: "bfy__videoshow__header",
+        videoPlayer: {
+            content: "bfy__videoshow__videoPlayer-content"
         },
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/Videoshow.jsx:27]: video-player

File: src/Pages/Videoshow/Videoshow.jsx [hunk 20]
```diff
@@ -1,58 +1,54 @@
...
-   const getSidebars = () => (
-    <Sidebars />
-    )
-    const getVideoPlayer = () => (
-    <div className={styles.video_Player.content}> 
-        {videos.map((item) =>(<VideoshowView videos={item} videoId={videoId} key={item._id} />  ))}
-    </div> 
+    const getVideoPlayerView = () => (
+        <div className={styles.videoPlayer.content}>
+            {videos.map((item) => (<VideoshowView videos={item} videoId={videoId} key={item._id} />))}
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/Videoshow.jsx]: remove inline

File: src/Pages/Videoshow/VideoshowView.jsx [hunk 21]
```diff
@@ -1,119 +1,116 @@
...
+                        </span>
+                        <span className={styles.videoPlayer.button} onClick={() => playlistHandler()}><MdPlaylistAdd className='bars-icon' />
+                        </span>
                     </div>
+                </div>
+                <div className={styles.videoPlayer.discription}>
+                    <p className='margin0 Font'>{publishedAt} </p>
+                    <br />
+                    <p className='margin0 Font'>{isExpanded ? description : description.slice(0, 100) + "..."}</p>
+                    <h4 className={styles.videoPlayer.showButton} onClick={toggleExpanded}>{isExpanded ? "Show Less" : "Show More"}</h4>
```
Comment (agrawalshanu708) [inline src/Pages/Videoshow/VideoshowView.jsx:104]: use usememo here

File: src/Components/PlaylistModal/Playlistmodal.jsx [hunk 22]
```diff
@@ -1,59 +1,64 @@
...
-function PlaylistModal( videos ) {
-    const {playlistState , playlistDispatch } = usePlaylist()
+function PlaylistModal(videos) {
+    const { playlistState, playlistDispatch } = usePlaylist()
     const { Playlistitem } = playlistState
-    const [name ,setName] = useState('')
-    const { Newvideo} = videos
-    console.log(Playlistitem)
+    const [name, setName] = useState('')
+    const { Newvideo } = videos
```
Comment (agrawalshanu708) [inline src/Components/PlaylistModal/Playlistmodal.jsx:12]: gfgvdz

**Review by agrawalshanu708 — CHANGES_REQUESTED** [review]

File: src/Pages/Videoshow/Videoshow.jsx (hunk 18 above)
Comment (Somil680) [reply src/Pages/Videoshow/Videoshow.jsx:22]: done

**Pull request merged**